importer/src/path_canonicalizer.rs

```diff
@@ -0,0 +1,378 @@
+/// Replace all relative paths in the program with absolute paths to the canonical symbol they point to, and remove all import statements in the program
+use number::FieldElement;
+
+use std::collections::BTreeMap;
+
+use ast::parsed::asm::{
+    ASMModule, ASMProgram, AbsoluteSymbolPath, Import, Machine, MachineStatement, Module,
+    ModuleStatement, Symbol, SymbolDefinition, SymbolPath,
+};
+
+pub fn canonicalize_paths<T: FieldElement>(
+    program: ASMProgram<T>,
+) -> Result<ASMProgram<T>, String> {
+    let paths = generate_path_map(&program)?;
+
+    let program = ASMProgram {
+        main: canonicalize_paths_in_module(AbsoluteSymbolPath::default(), program.main, &paths),
+    };
+
+    Ok(program)
+}
+
+// replace references to symbols with absolute paths. This removes the import statements. This always succeeds if the symbol table was generated correctly.
+fn canonicalize_paths_in_module<T: FieldElement>(
+    absolute_path: AbsoluteSymbolPath,
```

Comment:
Is the the absolute path to the module? Then maybe `module_path` would be more suitable, we already know that it is an absolute path from it's type.